character-sfx.js

```diff
@@ -188,29 +193,37 @@ class CharacterSfx {
     _handleStep();
 
     const _handleNarutoRun = () => {
-      if(ioManager.keys.right){
-        this.currentDirectionState='right';
-      }   
-      if(ioManager.keys.left){
-        this.currentDirectionState='left';
-      }  
-      if(ioManager.keys.up){
-        this.currentDirectionState='up';
-      }   
-      if(ioManager.keys.down){
-        this.currentDirectionState='down';
-      }
+      
+      this.player.getWorldDirection(localVector)

```

Comment:
This is slow because it does a matrix update.

Can use `Vector3.setFromMatrixPosition(this.player.matrixWorld)` instead.

https://threejs.org/docs/#api/en/math/Vector3.setFromMatrixPosition